bfvmm/src/vmcs/src/vmcs_intel_x64_check_controls.cpp

```diff
@@ -77,7 +79,16 @@ vmcs_intel_x64::check_control_ctls_reserved_properly_set(uint64_t msr_addr, uint
         throw std::logic_error(std::string("invalid ") + name);
     }
 
-    if ((ctls & ~allowed1) != 0UL)
+    allowed1_failed = (ctls & ~allowed1) != 0UL;
+
+    if (msrs::ia32_vmx_procbased_ctls2::addr == msr_addr)
+    {
+        using namespace vmcs::primary_processor_based_vm_execution_controls;
+
```

Comment:
Extra while space is not needed, here. Also `bool allowed1_failed = false` is redundant. Prefer `auto allowed1_failed = false`, and I think the line below can be simplified with `&=`, and if we are doing a ton of `using namespace` stuff, we should just qualify those namespaces at the top of the file and call it a day 😄